### PR TITLE
Oleg_Mikus_mikus.oleg@gmail.com

### PR DESCRIPTION
**Pull request by OlegMikus**

### CODE REVIEW

File: final_task/rss_reader/json_date_util.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import json
+
+
+def json_convert(dict_with_info: dict, limit: str):
```
Comment (HenadziStantchik) [inline final_task/rss_reader/json_date_util.py]: (Optional) If you are using typehints it is better to specify return type too.

File: final_task/rss_reader/json_date_util.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+              'Entries': []
+              }
+    temp = {'Title': [],
+            'Date': [],
+            'Links': [],
+            'Info': []
+            }
+    if type(dict_with_info) == dict:
+        output['Feed'] = dict_with_info['Feed']
+        for index in range(int(len(limit))):
```
Comment (HenadziStantchik) [inline final_task/rss_reader/json_date_util.py]: During our sessions throughout the course we *emphasized multiple times* that you should __not__ use such constructions. Please fix it.

File: final_task/rss_reader/json_date_util.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    output = {'Feed': [],
+              'Entries': []
+              }
+    temp = {'Title': [],
+            'Date': [],
+            'Links': [],
+            'Info': []
+            }
+    if type(dict_with_info) == dict:
```
Comment (HenadziStantchik) [inline final_task/rss_reader/json_date_util.py:16]: It is better to use `isinstance` ~~of~~ `is` operator for type checks.

File: final_task/rss_reader/text_work_util.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import feedparser
+import html
+import logging
+
+
+def take_info(source: str, limit: str):
+    # take info from URL and create dict to storage news
```
Comment (HenadziStantchik) [inline final_task/rss_reader/text_work_util.py]: Here and below: Docstrings should be ike this: `"""Text"""`

File: final_task/rss_reader/text_work_util.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        for index in range(int(limit)):
+            print('\n' + 'Title: ' + dict_whit_info['Title'][index])
+            print('Date: ' + dict_whit_info['Date'][index])
+            print('Link: ' + dict_whit_info['Links'][index])
+            print('Info: ' + dict_whit_info['Info'][index])
+    else:
+        print(dict_whit_info)
+
+
+def make_text_better(text_of_news_list: list):
```
Comment (HenadziStantchik) [inline final_task/rss_reader/text_work_util.py]: Here, above and below: It is better to use more specific function names
Same for variable names

**Comment by HenadziStantchik** [comment]
Here is some feedback on your app:

1. if `--limit` is not specified user should get all available feed.

2. It is recommended to not throw out image links but have them in the news feed, this will be helpful in the future iterations.

3. `--json` works strange

4. if `--limit` is larger than feed then you get an exception with traceback in stdout

**Comment by OlegMikus** [comment]
> Here is some feedback on your app:
> 
> 1. if `--limit` is not specified user should get all available feed.
> 2. It is recommended to not throw out image links but have them in the news feed, this will be helpful in the future iterations.
> 3. `--json` works strange <- What does it mean?
> 4. if `--limit` is larger than feed then you get an exception with traceback in stdout

**Comment by OlegMikus** [comment]
> 3. `--json` works strange

What does it mean?

**Comment by HenadziStantchik** [comment]
> > 1. `--json` works strange
> 
> What does it mean?

If the limit is not specified, it will print 1 news entry. if the limit is ~1000 it will print 2 entries. if the --limit is ~124123127 it will print three entries. Please check its work

File: final_task/rss_reader/json_date_util.py (hunk 3 above)
Comment (HenadziStantchik) [reply final_task/rss_reader/json_date_util.py:16]: or*

File: final_task/rss_reader/rss_reader.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    parser.add_argument("--verbose", action="store_true", help="Outputs verbose status messages")
+    parser.add_argument("--limit", help="Limit news topics if this parameter provided",)
+
+    args = parser.parse_args()
+
+    return args
+
+
+def main_func():
+    limit = parsed().limit
```
Comment (HenadziStantchik) [inline final_task/rss_reader/rss_reader.py:21]: you don't really needd to call this function every time:
```
args = parsed()
limit  = args.limit
.....
```

File: final_task/rss_reader/rss_reader.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
+import argparse
+import json_date_util as util
+import text_work_util as dt
+
+
+def parsed():
```
Comment (HenadziStantchik) [inline final_task/rss_reader/rss_reader.py:6]: It is better to use variable and function names that contain unambiguous answer to the question: `What is this?`

File: final_task/rss_reader/text_work_util.py [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+    if limit:
+
+        try:
+            if int(limit) > len(all_info.entries) or int(limit) < 0:
+                limit = int(limit)
+            else:
+                limit = all_info.entries[0:int(limit)]
+        except ValueError:
+            print('\n Should try only integer numbers')
+            sys.exit()
```
Comment (HenadziStantchik) [inline final_task/rss_reader/text_work_util.py:17]: Here and below: It is better to have only one exit point from the application. For example you can raise your custom exception here or reraise this exception and handle it in the main function

File: final_task/rss_reader/text_work_util.py [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+
+    else:
+        limit = all_info.entries
+
+    try:
+        for enter in limit:
+            dict_with_info['Title'] += [enter.title]
+            dict_with_info['Date'] += [enter.published]
+            dict_with_info['Links'] += [enter.link]
+            dict_with_info['Info'] += [enter.summary]
```
Comment (HenadziStantchik) [inline final_task/rss_reader/text_work_util.py:27]: A question:

Do you really need to put objects inside lists here? Why not just simply use these objects?

`[enter.title]` and etc. 

UPD: if you want to add one element to an existing list it is better to use `your_list.append(new_element)`

**Comment by HenadziStantchik** [comment]
1. `--json` option still does not work:

mikus

2. If the limit is larger than feed size user should get all available news entries.

3. if option `--version` is specified by user, your app should print out version and then shutdown, without printing any news.

**Pull request closed without merging**